internal/mutator/pod/agent/injector.go

```diff
@@ -364,9 +364,27 @@ func (i *injector) shouldInjectContainer(ctx context.Context, pod *corev1.Pod, n
 		return false, fmt.Errorf("getting Namespace %q for policy matching: %w", namespace, err)
 	}
 
+	// Checking if there is any overlapping with the volumes we want to mount and the volumes already present.
```

Comment:
Hmm, I wouldn't put this check in `shouldInjectContainer`. In case of an error, we do not fail to _check_ if we should inject the container, but the received Pod does not meet the requirements for injection. 

I'd leave `shouldInjectContainer` as it is and add another helper, called `canInjectContainer`, to which we can later on add a check for service account token.

Reply:
make sense!